### PR TITLE
[controller] fix for NPE during HybridStoreConfigImpl creation

### PR DESCRIPTION
**Pull request by arjun4084346**
## Fix NPE in HybridStoreConfigImpl creation

This PR fixes the following NPE coming during the initialization of HybridStoreConfigImpl. Existing stores do not have the field `realTimeTopicName` and when we try to get such stores from zk, it assigns null value to that field. In this PR, I am setting it to a default non-null value.
 https://github.com/linkedin/venice/commit/cc6e3db6f898a0c5efd1aa18847511b113525ab5. PR https://github.com/linkedin/venice/pull/1345 introduced this issue.

`Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.toString()" because "this.hybridConfig.realTimeTopicName" is null at com.linkedin.venice.meta.HybridStoreConfigImpl.getRealTimeTopicName(HybridStoreConfigImpl.java:108) ~[com.linkedin.venice.venice-common-0.4.368.jar:?] at com.linkedin.venice.meta.HybridStoreConfigImpl.clone(HybridStoreConfigImpl.java:146) ~[com.linkedin.venice.venice-common-0.4.368.jar:?] at com.linkedin.venice.meta.ZKStore.<init>(ZKStore.java:186) ~[com.linkedin.venice.venice-common-0.4.368.jar:?] at com.linkedin.venice.meta.ZKStore.cloneStore(ZKStore.java:969) ~[com.linkedin.venice.venice-common-0.4.368.jar:?] at com.linkedin.venice.helix.HelixReadWriteStoreRepository.lambda$getAllStores$0(HelixReadWriteStoreRepository.java:96) ~[com.linkedin.venice.venice-common-0.4.368.jar:?] at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] at java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3612) ~[?:?] at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?] at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?] at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?] at com.linkedin.venice.helix.HelixReadWriteStoreRepository.getAllStores(HelixReadWriteStoreRepository.java:96) ~[com.linkedin.venice.venice-common-0.4.368.jar:?] at com.linkedin.venice.helix.HelixReadOnlyStoreRepositoryAdapter.getAllStores(HelixReadOnlyStoreRepositoryAdapter.java:127) ~[com.linkedin.venice.venice-common-0.4.368.jar:?] at com.linkedin.venice.controller.HelixVeniceClusterResources.repairStoreReplicationFactor(HelixVeniceClusterResources.java:215) ~[com.linkedin.venice.venice-controller-0.4.368.jar:?] at com.linkedin.venice.controller.HelixVeniceClusterResources.refresh(HelixVeniceClusterResources.java:244) ~[com.linkedin.venice.venice-controller-0.4.368.jar:?] at com.linkedin.venice.controller.VeniceControllerStateModel.initClusterResources(VeniceControllerStateModel.java:195) ~[com.linkedin.venice.venice-controller-0.4.368.jar:?] at com.linkedin.venice.controller.VeniceControllerStateModel.lambda$onBecomeLeaderFromStandby$0(VeniceControllerStateModel.java:150) ~[com.linkedin.venice.venice-controller-0.4.368.jar:?] at com.linkedin.venice.controller.VeniceControllerStateModel.executeStateTransition(VeniceControllerStateModel.java:111) ~[com.linkedin.venice.venice-controller-0.4.368.jar:?] at com.linkedin.venice.controller.VeniceControllerStateModel.onBecomeLeaderFromStandby(VeniceControllerStateModel.java:129) ~[com.linkedin.venice.venice-controller-0.4.368.jar:?] ... 12 more`

Resolves #XXX

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-common/src/test/java/com/linkedin/venice/helix/TestStoreJsonSerializer.java [hunk 1]
```diff
@@ -27,6 +29,49 @@
  * Test cases for StoreJsonSerializer.
  */
 public class TestStoreJsonSerializer {
+  @Test
+  void testRealTimeTopicNameDefault() throws IOException {
+    File testFile = new File("src/test/resources/testHybridStore.json");
```
Comment (nisargthakkar) [inline internal/venice-common/src/test/java/com/linkedin/venice/helix/TestStoreJsonSerializer.java:34]: Can we read the file through `Thread.currentThread().getContextClassLoader().getResourceAsStream(fileName)` ?

File: internal/venice-common/src/test/resources/testHybridStore.json
Comment (nisargthakkar) [inline internal/venice-common/src/test/resources/testHybridStore.json:1]: Can you put these files in a dedicated directory with the name denoting the purpose?

File: internal/venice-common/src/test/resources/testHybridStore.json [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+          "minActiveReplicas": 2,
+          "useVersionLevelIncrementalPushEnabled": true,
+          "useVersionLevelHybridConfig": true,
+          "dataRecoveryVersionConfig": null,
+          "repushSourceVersion": -1,
+          "chunkingEnabled": true,
+          "incrementalPushEnabled": false,
+          "separateRealTimeTopicEnabled": false,
+          "nativeReplicationEnabled": true,
+          "pushStreamSourceAddress": "kafka.venice.kafka.prod-lva1.atd.prod.linkedin.com:16637",
```
Comment (huangminchn) [inline internal/venice-common/src/test/resources/testHybridStore.json:77]: Thanks for adding the test! However, I don't think we can expose LI specific urls to open source.

**Review by huangminchn — CHANGES_REQUESTED** [review]
Thanks for quick turnaround Arjun! It's okay to leave stacktrace in PR description, but when we squash and commit, please remove them or just keep a few lines to keep the commit history precise.

File: internal/venice-common/src/test/java/com/linkedin/venice/schema/SchemaAdapterTest.java [hunk 3]
```diff
@@ -26,7 +26,7 @@ public class SchemaAdapterTest {
   private static final Schema EVOLVED_RECORD_SCHEMA = AvroSchemaParseUtils
       .parseSchemaFromJSONStrictValidation(loadFileAsString("AvroRecordUtilsTest/EvolvedRecordSchema.avsc"));
 
-  private static String loadFileAsString(String fileName) {
+  public static String loadFileAsString(String fileName) {
```
Comment (nisargthakkar) [inline internal/venice-common/src/test/java/com/linkedin/venice/schema/SchemaAdapterTest.java:29]: Can you please put it in a shared util class if we want to reuse this? Making random classes depend on a function from `SchemaAdapterTest` is not clean

**Comment by arjun4084346** [comment]
> Thanks for quick turnaround Arjun! It's okay to leave stacktrace in PR description, but when we squash and commit, please remove them or just keep a few lines to keep the commit history precise.

oh, so PR description , and not just commit message, goes into the history! will remove it during squash

File: internal/venice-test-common/src/main/java/com/linkedin/venice/utils/TestUtils.java [hunk 4]
```diff
@@ -80,18 +80,9 @@
...
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-import java.util.Properties;
-import java.util.TreeMap;
+import java.util.*;
```
Comment (nisargthakkar) [inline internal/venice-test-common/src/main/java/com/linkedin/venice/utils/TestUtils.java:85]: Can you revert the wildcard imports? You might need to update your intellij settings. (There's a bug in IntelliJ that these get reverted on each restart)

**Review by nisargthakkar — APPROVED** [review]
Thanks. LGTM

**Pull request merged**